src/sqlfluff/core/parser/segments/base.py

```diff
@@ -842,42 +914,100 @@ def realign(self):
+            pos_marker=self.pos_marker,
+        )
 
-        while True:
-            if len(todo_buffer) == 0:
-                # We're done.
-                break
-            else:
-                # Get the first off the buffer
-                seg = todo_buffer.pop(0)
-
-                # We'll preserve statement indexes so we should keep track of that.
-                # When recreating, we use the DELTA of the index so that's what matter...
-                idx = seg.pos_marker.statement_index - running_pos.statement_index
-                if seg.is_meta:
-                    # It's a meta segment, just update the position
-                    seg = seg.__class__(pos_marker=running_pos)
-                elif len(seg.segments) > 0:
-                    # It's a compound segment, so keep track of it's children
-                    child_segs = seg.segments
-                    # Create a new segment of the same type with the new position
-                    seg = seg.__class__(segments=child_segs, pos_marker=running_pos)
-                    # Realign the children of that class
-                    seg = seg.realign()
-                else:
-                    # It's a raw segment...
-                    # Create a new segment of the same type with the new position
-                    seg = seg.__class__(raw=seg.raw, pos_marker=running_pos)
-                # Update the running position with the content of that segment
-                running_pos = running_pos.advance_by(raw=seg.raw, idx=idx)
-                # Add the buffer to my new segment
-                seg_buffer.append(seg)
+    def iter_patches(self, templated_str: str) -> Iterator[FixPatch]:
+        """Iterate through the segments generating fix patches.
 
-        # Create a new version of this class with the new details
-        return self.__class__(segments=tuple(seg_buffer), pos_marker=self.pos_marker)
+        The patches are generated in TEMPLATED space. This is important
+        so that we defer dealing with any loops until later. At this stage
+        everything *should* happen in templated order.
+
+        Occasionally we have an insertion around a placeholder, so we also
+        return a hint to deal with that.
+        """
+        # Does it match? If so we can ignore it.
+        matches = self.raw == templated_str[self.pos_marker.templated_slice]
+        if matches:
+            return
+
+        # If we're here, the segment doesn't match the original.
+
+        # If it's all literal, then we don't need to recurse.
+        if self.pos_marker.is_literal:
+            # Yield the position in the source file and the patch
+            patch = FixPatch(self.pos_marker.templated_slice, self.raw)
+            linter_logger.debug("Yeilding literal patch: %r", patch)
+            yield patch
+        else:
+            # This segment isn't a literal, but has changed, we need to go deeper.

```

Comment:
I'm with you <img src="https://i.kym-cdn.com/entries/icons/mobile/000/012/886/wntgd.jpg" height=150 />

Reply:
I thought someone might enjoy that reference 😆